[Keycloak CI] - Cookies tests - KcOidcBrokerTokenExchangeTest

### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

ci

### Describe the bug

```
2025-01-10 09:56:22  -- [ERROR] org.keycloak.testsuite.broker.KcOidcBrokerTokenExchangeTest.testExternalInternalTokenExchange
2025-01-10 09:56:22  -- [ERROR]   Run 1: KcOidcBrokerTokenExchangeTest.testExternalInternalTokenExchange:84->assertExternalToInternalExchange:143 
2025-01-10 09:56:22  -- Expected: <200>
2025-01-10 09:56:22  --      but: was <400
...
2025-01-10 09:56:22  -- [ERROR] org.keycloak.testsuite.broker.KcOidcBrokerTokenExchangeTest.testExternalInternalTokenExchangeUsingIssuer
2025-01-10 09:56:22  -- [ERROR]   Run 1: KcOidcBrokerTokenExchangeTest.testExternalInternalTokenExchangeUsingIssuer:91->assertExternalToInternalExchange:143 
2025-01-10 09:56:22  -- Expected: <200>
2025-01-10 09:56:22  --      but: was <400>
```

https://keycloak-jenkins/view/Keycloak/job/keycloak-continuous-testing/61/execution/node/232/log/


### Version

latest

### Regression

- [ ] The issue is a regression

### Expected behavior

-

### Actual behavior

-

### How to Reproduce?

-

### Anything else?

_No response_

--- a/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/broker/KcOidcBrokerTokenExchangeTest.java
+++ b/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/broker/KcOidcBrokerTokenExchangeTest.java
@@ -69,6 +69,7 @@ import org.keycloak.testsuite.arquillian.annotation.EnableFeatures;
 import org.keycloak.testsuite.updaters.IdentityProviderAttributeUpdater;
 import org.keycloak.testsuite.util.AdminClientUtil;
 import org.keycloak.testsuite.util.OAuthClient;
+import org.keycloak.testsuite.util.ServerURLs;
 import org.keycloak.util.BasicAuthHelper;
 
 @EnableFeatures({@EnableFeature(Profile.Feature.TOKEN_EXCHANGE), @EnableFeature(Profile.Feature.ADMIN_FINE_GRAINED_AUTHZ)})
@@ -111,6 +112,14 @@ public final class KcOidcBrokerTokenExchangeTest extends AbstractInitializedBase
 
         RealmResource consumerRealm = realmsResouce().realm(bc.consumerRealmName());
         IdentityProviderResource identityProviderResource = consumerRealm.identityProviders().get(bc.getIDPAlias());
+
+        // if auth.server.host != auth.server.host2 we need to update the issuer in the IDP config
+        IdentityProviderRepresentation representation = identityProviderResource.toRepresentation();
+        if (!representation.getConfig().get("issuer").startsWith(ServerURLs.getAuthServerContextRoot())) {
+            representation.getConfig().put("issuer", ServerURLs.getAuthServerContextRoot() + "/auth/realms/provider");
+            identityProviderResource.update(representation);
+        }
+
         identityProviderResource.addMapper(hardCodedSessionNoteMapper).close();
 
         OAuthClient.AccessTokenResponse tokenResponse = oauth.doGrantAccessTokenRequest(bc.providerRealmName(), bc.getUserLogin(), bc.getUserPassword(), null, brokerApp.getClientId(), brokerApp.getSecret());
@@ -123,7 +132,7 @@ public final class KcOidcBrokerTokenExchangeTest extends AbstractInitializedBase
         Client httpClient = AdminClientUtil.createResteasyClient();
 
         try {
-            WebTarget exchangeUrl = httpClient.target(OAuthClient.AUTH_SERVER_ROOT)
+            WebTarget exchangeUrl = httpClient.target(ServerURLs.getAuthServerContextRoot() + "/auth")
                     .path("/realms")
                     .path(bc.consumerRealmName())
                     .path("protocol/openid-connect/token");

